### DISCUSSION TITLE
How to build and design blocks for consistent dev and user experiences

### DISCUSSION THREAD

**Discussion by justintadlock** [WordPress/developer-blog-content#346] [Topic Ideas]
We should have a good guide for building blocks so that they:

- Provide a nice user experience that is consistent with Core design principles
- Make it easy for theme authors to integrate with them
- Favors smaller pieces that can be integrated into patterns
- Integrate with Core design tools
- Use Core components as much as possible
- When a binding, variation, or style variation would be better

Basically, this should be a sort of "dos" and "don'ts" of block building with perhaps some tips/tricks thrown in. Mostly, I'm just throwing out some ideas at the moment. Please contribute to the discussion.

This idea came out of a discussion around WooCommerce and creating a consistent user experience. CC: @ellenbauer

**Comment by troychaplin** [comment]
Maybe a bit off topic, but what about a series of articles that ties into this initiative where each month a commonly used design is taken and someone tries to recreate it as close as possible using only the available core blocks.

**Reply by justintadlock** [reply]
Yep, definitely a different topic. But I could definitely see exploring that as a series. I used to do something similar back when I wrote for WP Tavern.

**Comment by troychaplin** [comment]
If the idea here is to go beyond building block using available core blocks there could be a series of external templates creating with common examples that give folks a good starting point that already meet existing standards, leverage some of the available components, etc

**Reply by justintadlock** [reply]
The goal here is define consistent UI and UX standards. A future goal beyond that could be to create those templates based on whatever these standards are.

**Comment by bph** [comment]
topic approved. conversation continues on this issue: 
- #350
